firestore/apiv1/firestore_client.go

```diff
@@ -734,6 +735,17 @@ type gRPCClient struct {
 	xGoogMetadata metadata.MD
 }
 
+// requestParamsHeader is routing header required to access named databases
```

Comment:
@viacheslav-rostovtsev can you PTAL?

Reply:
This seems fine, and as a bonus it would work with something like `"projects/*/databases/*/documents/**"` (`parent` in `CreateDocument` rpc).
You're making an assumption that project IDs and database IDs will never contain symbols that need to be %-encoded. According to RDC6570, this is not true for e.g. the `(` and `)` symbols in `(default)` (and I don't know what other symbols can be in the database id). If the backend is fine with it, it's OK to leave it as is but consider documenting what's going on.

Reply:
Added escaping